src/Domain/JacobianDiagnostic.cpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/JacobianDiagnostic.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace domain {
+template <size_t Dim, typename Fr>
+void jacobian_diagnostic(
+    const gsl::not_null<
+        tnsr::i<DataVector, Dim, typename Frame::ElementLogical>*>
+        jacobian_diag,
+    const Jacobian<DataVector, Dim, typename Frame::ElementLogical, Fr>&
+        analytic_jacobian,
+    const TensorMetafunctions::prepend_spatial_index<
+        tnsr::I<DataVector, Dim, Fr>, Dim, UpLo::Lo,
+        typename Frame::ElementLogical>& numeric_jacobian_transpose) {
+  const size_t number_of_points = analytic_jacobian.begin()->size();
+  destructive_resize_components(jacobian_diag, number_of_points);
+
+  // i_hat = logical frame index
+  // i = mapped frame index
+  for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {
+    jacobian_diag->get(i_hat) = -abs(analytic_jacobian.get(0, i_hat));
+    for (size_t i = 1; i < Dim; ++i) {
+      jacobian_diag->get(i_hat) -= abs(analytic_jacobian.get(i, i_hat));
+    }
+    Scalar<DataVector> jacobian_component_sum{number_of_points};
```

Comment:
Uh oh, memory allocation in a loop! Can you go back to allocating outside the loop. Just move this line to line 30, nothing else needs to be changed!

Reply:
🤦‍♂️that's what I get for trying to get it in 5 minutes before lunch! "O, let's declare this when we need it!", I thought.

Anyway, thanks for catching, fixed!

The test failures look python related, not related to my PR?